recipes/server_check.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_relative './server_check_helper'
+
+consul_service_file = '/etc/systemd/system/consul.service'
+consul_service_folder = '/etc/systemd/system/consul.service.d'
+drop_in_file = "#{consul_service_folder}/server_start_prevention.conf" # prevent systemd from starting
+
+directory consul_service_folder do
+  owner 'root'
+  mode '0744'
+  action :create
+end
+
+# Output file is used for systemd to check whether it should start consul.
+if is_rogue_server?(node.default['consul']['config']['retry_join'], node['ipaddress'])
+  # Stop the service if it is already running.
+  service 'consul' do
+    action :stop
+    only_if { ::File.exist?(consul_service_file) }
+  end
+
+  # Update config itself to further prevent it from becoming a server.
+  node.override['consul']['config']['server'] = false
+
+  # Currently has to use ConditionPathExists instead of ExecCondition.
+  # Blocks the service from starting if this file itself is present.
+  # ExecCondition requires systemd 243+, Ubuntu 1804 has systemd 237.
+  file drop_in_file do
+    owner 'root'
+    mode '0744'
```

Comment:
Does the conf file need user execute bit?

```suggestion
    mode '0644'
```

Reply:
The user doesn't need to execute this, this is automatically picked up by systemd to check for condition before it starts a service.

Reply:
> The user doesn't need to execute this

Exactly, so can we drop the +x bit? Change 7 to 6

Reply:
Resolved in https://github.com/Roblox/consul-cookbook/pull/44/commits/a3210fa5d0d7047e244e6446f47c90bc1ccd1a90